### PR TITLE
Add caster-attribute scaling for skill effects (#741)

### LINKED ISSUE #741: Add the ability for skill effects to scale with attributes
For example.

Allow a skill that applies DamageTakePerSecond to the enemy to have the magnitude scale with the Dexterity stat.

### PR DESCRIPTION
**Pull request by ginderjeremiah**
## Summary

Skill **damage** already scales with attributes (`BaseDamage + Σ(attribute × multiplier)`), but skill **effects** used a flat authored `Amount`. This adds the same capability to effects: an effect's magnitude can now scale off one of the **caster's** attributes.

The applied magnitude is `Amount + casterAttribute × ScalingAmount`, read from the **casting** battler regardless of which battler the effect lands on — exactly mirroring how a `DamageMultiplier` reads the caster. So the issue's example works directly: a poison applying `DamageTakenPerSecond` to the enemy can scale with the player's Dexterity. A `ScalingAmount` of `0` means no scaling, so **existing effects are byte-identical** (the parity replay is unaffected).

Per the design decision, V1 supports a **single scaling attribute per effect** (two columns, no grandchild table); multi-attribute scaling is a clean later follow-up.

## How it addresses the issue

- **Data model** — `ScalingAttributeId` (`EAttribute`) + `ScalingAmount` added to `SkillEffect` across the entity, domain model, read contract, mapper, and admin persistence, with an additive EF migration (columns default to `0` = no scaling).
- **Battle runtime (parity-critical, both sides)** — `BattleContext.ApplySkillEffect` (backend) and `Skill.applyEffects` (frontend) resolve the magnitude from the active/casting battler and pass it to `Battler.ApplyEffect`. The shared frontend formula lives in `battle-formulas.ts` (consumed by both the simulation and the display surfaces). The magnitude is resolved once at apply time and frozen on the modifier.
- **Authoring** — a "Scales with" attribute column and a "Per point" coefficient column in the Workbench skill-editor Effects table (and the save-diff equality now includes them so a scaling-only edit persists).
- **Display** — the active-effect chips, the combat log, and the skill tooltip all show the resolved (scaled) magnitude.
- **Codegen** — regenerated the TypeScript client (`ISkillEffect` gains the two fields).
- **Docs** — updated the Skill Effects section in `game-design.md` and the runtime section in `backend-battle.md`.

## Test plan

- [x] Backend unit tests: new `BattlerEffectsTests` scaling cases (scales with caster, reads caster-not-target, zero-scaling no-op).
- [x] **Mirrored parity scenario** `effectScalesWithCasterIntellect` added to both `BattleSimulatorParityTests.cs` and `battle-simulation-parity.test.ts` (a poison whose authored amount is 0 scales to 50 DTPS via the caster's Intellect, killing the enemy at 2960ms — and dealing nothing if scaling regresses).
- [x] Frontend unit tests mirroring the backend scaling cases, plus chip/log/tooltip resolved-magnitude coverage.
- [x] Admin integration test: scaling round-trips through persistence.
- [x] Full backend suite green (Core 625, Infrastructure 42, Application 330, Api 526).
- [x] Full frontend suite green (2124 tests); `eslint` + `svelte-check` clean.

## Follow-ups

- Multi-attribute effect scaling (promote the single field to a collection, mirroring `DamageMultiplier`) if content ever needs an effect to scale off more than one attribute.

Closes #741

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_01M44B24WkfsFYhTHuBvmy18)_

### CODE REVIEW

File: Game.Core.Tests/Battle/BattlerEffectsTests.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public void ApplyEffect_AddsModifier_RaisesAttribute()
         {
             var battler = MakeBattler(Stat(Strength, 10));
 
-            battler.ApplyEffect(Effect(1, Strength, Additive, 5));
+            Apply(battler,Effect(1, Strength, Additive, 5));
```
Comment (ginderjeremiah) [inline Game.Core.Tests/Battle/BattlerEffectsTests.cs]: **Blocking (CI red):** `Apply(battler,Effect(...))` is missing the space after the comma. `dotnet format --verify-no-changes` flags this on 12 lines here (24, 35, 36, 48, 49, 61, 82, 86, 102, 114, 127, 128: `error WHITESPACE: ... Insert '\s'`), which fails the `test-backend` job *before any tests run* — so the "backend suite green" note in the description came from a local run that skipped the formatter. Fix is mechanical: `dotnet format` (or add the spaces: `Apply(battler, Effect(...))`). Everything else in the backend looks good.

---
_Generated by [Claude Code](https://claude.ai/code)_

**Review by ginderjeremiah — COMMENTED** [review]
## Review: Caster-attribute scaling for skill effects (#741)

**Verdict: Request changes** — the change itself is excellent and ready, but `test-backend` is currently **red** on the `dotnet format` gate, so it can't merge as-is. (Posting as a comment since I'm the PR author and GitHub won't accept a gating review on my own PR; the real verdict is request-changes until CI is green.)

### Blocking

- **CI: `dotnet format --verify-no-changes` fails** — `Game.Core.Tests/Battle/BattlerEffectsTests.cs` introduces `Apply(battler,Effect(...))` with no space after the comma on 12 lines (24, 35, 36, 48, 49, 61, 82, 86, 102, 114, 127, 128). The format step runs before the test step, so the `test-backend` job exits 2 and the suite never executes — the "backend suite green" line in the description reflects a local run that didn't include the formatter. One `dotnet format` pass fixes it. No other CI job has reported a failure (chromium e2e is green; webkit/firefox/frontend still in progress at review time).

### What's good

This is a genuinely thorough, well-scoped implementation — no other findings:

- **Parity is real and guarded.** The magnitude formula is defined once per side (`scaledEffectAmount` in `battle-formulas.ts` ↔ `BattleContext.ApplySkillEffect`) and computed identically (`amount + casterAttr × scalingAmount`). The mirrored `effectScalesWithCasterIntellect` parity scenario, with its hand-computed 2960ms kill, locks the two implementations together and would fail loudly if scaling regressed on either side.
- **Caster-not-target semantics are correct and tested** on both sides (`ScalingReadsCasterNotTarget` / "reads the caster attribute, not the target"), and the resolved magnitude is frozen on the modifier at apply time rather than re-evaluated — matching how `DamageMultiplier` reads the caster.
- **Backward compatibility is clean** — `ScalingAmount = 0` ⇒ byte-identical behavior, additive migration with `defaultValue: 0`, both FKs configured explicitly so EF's convention doesn't have to disambiguate the two `Attribute` nav pairs.
- **Layering is respected** — scaling resolves in the domain (`BattleContext`), the mapper/admin-persistence/codegen all carry the two new fields, and the display surfaces (chips, combat log, tooltip) all render the resolved magnitude with a sensible unscaled fallback when no caster is available.
- **Test coverage meets the bar** — backend unit + frontend mirror, admin round-trip integration test, display-helper coverage, and the cross-impl parity case. Docs updated in both `game-design.md` and `backend-battle.md`.

### Test coverage

Sufficient and well-aligned with the project's rules — domain logic unit-tested with edge cases (scales / reads-caster / zero-scaling), battle logic mirrored frontend↔backend including the parity scenario, persistence covered by an integration test. The only gap is operational: the failing format gate means the backend tests didn't actually run in CI this round, so green needs to be confirmed after the formatting fix.

Once `dotnet format` is applied and `test-backend` (plus the in-flight frontend/e2e jobs) goes green, this is an approve.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: Game.Core.Tests/Battle/BattlerEffectsTests.cs (hunk 1 above)
Comment (ginderjeremiah) [reply Game.Core.Tests/Battle/BattlerEffectsTests.cs]: Fixed in 6f6844a — ran `dotnet format`, which restored the missing space (`Apply(battler, Effect(...))`). `dotnet format --verify-no-changes` now passes on the full solution, and the backend tests still pass locally. The new CI run picked up the fix and `test-backend` is re-running.

---
_Generated by [Claude Code](https://claude.ai/code)_
